### ISSUE TITLE
Suppliers can edit their products on a hub's order

### ISSUE THREAD

**Issue by tschumilas** [openfoodfoundation/openfoodnetwork#5750]
## Description
As a supplier,  I am currently able to edit my products and add/delete my products in an order cycle created by a hub I sell through.  And,  I can get a report on my products' sales through the hub.     BUT I cannot add or delete my products to an order received by the hub.

As a supplier,  I want to be able to add or delete my products on an order received by a hub I sell through.   

Use case:   Hubs are moving to contact-less pickups and are under-resourced to do this.  Most hubs are asking suppliers to do more of the packing and order organizing.   Today,  as a supplier of variable weight items,  I have to write a list of all the weights of the products,  and send these to the hub,   who needs to enter them into each customer's order.   It would be more efficient if, 
 as the supplier,  I could enter the weight directly into order using BOM as I weight out the items.

Use case:  Suppliers will soon be able (with the hub's permission)  to pre-pack orders by customer.  But today,  if something I offered is not available,  I must tell this to the hub,  who has to have someone change the orders.   It would be more efficient if, 
 as the supplier,  I could delete any of my items that are not available using BOM  myself.
In many situations,  hubs are under-staffed and are asking suppliers to pre-pack orders.

## Acceptance Criteria & Tests

The supplier should only be able to edit, add, delete their OWN products in BOM.
The supplier should not be able to make other adjustments to orders,  or change/view payment status, etc.  of an order.

1.Supplier can adjust the weight of a variable weight item -  for their products only -  using BOM
2. Supplier can delete one of their items from a hub's order  - using BOM
3. Supplier can add a new item item to an order - using BOM.

**Comment by RachL** [comment]
hello @tschumilas :) Was there a discussion on Slack where this was considered small enough to be a potential papercut? It seams to me that it can be a big tasks, so it should go on to discourse first. But maybe I've missed something?

**Comment by tschumilas** [comment]
This was not discussed on slack as a papercut.  It was an idea that @kirstenalarsen suggested when we were slacking about challenges farmers markets had,  given we don't do anything like split payments. Its a way to ease the load carried by order cycle 'coordinators'  (informal farmers markets).    Sorry for my error -  I'm still not understanding when something comes to git as an enhancement and when it goes to discourse.  (Maybe we want to remove the possibility of an enhancement label in github to 'force' new features/improvements to go to discourse first?  If thats what we want?)

**Comment by Erioldoesdesign** [comment]
I think this came to the a right place (not everything needs to be discourse first and can be discoursed after an issue is made)

I understand the logic of what you want to do and why and it's importance. I can't speak for the dev side of the size of the task but from a UX pov, is there any need for a way for the hubs that you supply to receive a notification (of some kind) if you delete a product that is no longer available through BOM?

Basically is the information trail serviced already through OFN so everybody who needs to know what is being edited/deleted is getting what they need? If so then no additional UX but if we have questions or concerns then there will be some UX digging for this one.

**Comment by RachL** [comment]
No information trail today... Even worse: no trail at all. That's what the compliance on invoices features is suppose to solve.

If I speak for FR, the most important thing is for the trail to be there (want was edited and when), not to get en email each time a supplier changes something (when you have 80 suppliers e.g. this becomes too much in terms of number of notifications).

**Comment by tschumilas** [comment]
Right now - only the distributor (the seller:  farm shop, hub shop or distributor shop in an OC) is the only one who can access the orders,  edit the orders.  No one knows what changes were made.  I edit orders every week -  and even I don't know what changes were made and can't respond when buyers ask.  This is very bad.

When we do the compliant invoices issue -  then the changes will be recorded on the order (and the invoice?).  Anyway - if its on the ordre -  the seller can send a new confirmation to the buyer to show them the changes...  But nothing in the compliant invoices issue will cue the supplier to changes on the order.   The supplier will know only what their final products sold were  -  because they can run a supplier totals report....

This is not an issue when the supplier is the OC coordinator for their own shop --- they see and edit these orders.

The issue proposed here -  is to give suppliers to a hub shop access to their own product sales on orders.  So the supplier could delete, change or add their own products.
